### ISSUE TITLE
sceneReadyCallback returns null error when there is an error

### ISSUE THREAD

**Issue by dgngulcan** [tangrams/tangram-es#1778]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**
load a scene with an invalid URL

**RESULT:**
On console, exception is printed 
`D/OkHttp: java.io.IOException: Unexpected response code: Response{protocol=http/1.1, code=404, message=Not Found...`
 
sceneReadyCallback returns a null `Error` object with an integer sceneId

**EXPECTED RESULT:**
since the URL gives 404, I would expect an error of `MapController.Error` with `sceneReadyCallback`

**ENVIRONMENT:**
Android

**Comment by polar** [comment]
I can confirm this issue.

**Comment by matteblair** [comment]
Yep this error reporting facility is currently a little bit .. inadequate haha

I believe this error parameter was added specifically for scene updates - which reload the scene and invoke the scene ready callback. The current interface isn't well-suited for capturing all the things that could go wrong when loading a scene. To make things trickier, scene loading happens in native code so we can't just use exceptions. Our strategy so far has been to just write those kinds of errors to the log.
